docs/development_environment.md

```diff
@@ -0,0 +1,227 @@
+    "python.testing.pytestArgs": [],
+    "python.testing.unittestEnabled": false,
+    "python.testing.nosetestsEnabled": false,
+    "python.testing.pytestEnabled": true
+}
+```
+
+This will enable the python extension, debugging, linting and unit testing facilities.
+
+### 5.3 Run Config
+
+Going one step further, we can setup some default run configurations so that when `F5` is pressed, the server and is debuggable from within the editor. To crate a run config to start the server, in the `.vscode` directory, create a new file called `launch.json` with the following contents:
+```json
+{
+    "version": "0.2.0",
+    "configurations": [
+        {
+            "name": "Python: Flask",
+            "type": "python",
+            "request": "launch",
+            "module": "flask",
+            "env": {
+                "FLASK_APP": "elpis",
+                "FLASK_ENV": "development",
+                "FLASK_DEBUG": "0"
+            },
+            "args": [
+                "run",
+                "--host","0.0.0.0",
+                "--port","5000",
+                "--no-debugger",
+                "--no-reload"
+            ],
+            "jinja": true
+        }
+    ]
+}
+```
+
+Now press `F5` (or the play button) and see the server run. Try using breakpoints to pause the server at a line of code. Try the `elpis/elpis/__init__.py` file pause the program around line 72 on the print statement in the index function. To get the server to pause you will need to open a browser and load `0.0.0.0:5000` to run that section of code.
+
+## 5.4 Terminal
+
+Notice that when you press ctrl-` the terminal that opens is in the container.
+
+## 6. Elpis-GUI
+
+If you wish to develop the front end code along with the server, you can add an additional mount and override the `/elpis-gui` directory. Notice the second mount in the following code and replace `/path/to/elpis-gui` with the path to the `elpis-gui` git repository.
+
+```json
+{
+    "name": "Elpis Dev Container (in progress)",
+    "image": "coedl/elpis:latest",
+    "workspaceFolder": "/elpis",
+    "mounts": [
+        "type=bind,source=${localWorkspaceFolder},target=/elpis",
+        "type=bind,source=/path/to/elpis-gui,target=/elpis-gui"
+    ],
+    
+	"settings": { 
+		"terminal.integrated.shell.linux": "/bin/zsh",
+		"python.pythonPath": "/venv/bin/python",
+		"python.linting.pylintEnabled": true,
+		"python.linting.pylintPath": "/venv/bin/pylint",
+		"python.linting.enabled": true
+	},
+
+	"extensions": [
+		"ms-python.python"
+	]
+}
+```
+
+### 6.1 Elpis-GUI w/ Hot-Reload
+
+A more advanced method for developing the `elpis-gui` is to use the Webpack Dev Server's hot reload feature to automatically push your changes rather than waiting for `npm run watch` to pickup and then you having to force reload.
+
+To use hot reload:
+
+1. Update your `devcontainer.json` to include
+
+```json
+"forwardPorts": [
+    3000, // Forward the default webpack dev server port
```

Comment:
JSON does not support comments. Don't make me revoke your plumbing certification.